tests/flow/test_degree.py

```diff
@@ -0,0 +1,350 @@
+            (b)-[:S]->(a1),
+            (b)-[:S]->(b),
+            (b)-[:S]->(c),
+            (b)-[:S]->(d),
+
+            (c)-[:R]->(d)
+        """)
+
+    def test_invalid_invocation(self):
+        queries = ["CALL algo.degree()",                              # missing mandatory argument
+                   "CALL algo.degree(1)",                             # wrong argument
+                   "CALL algo.degree({dir: 'outgoing', unknown: 4})", # unexpected key
+                   "CALL algo.degree({dir: '4})"                      # invalid direction value
+        ]
+
+        for q in queries:
+            try:
+                self.graph.query(q)
+                self.env.assertFalse(True)
+            except Exception:
+                pass
+
+    def test_all_nodes_all_edges_degree(self):
+
+        # by default:
+        # 1. all nodes are considered
+        # 2. all edges are considered
+        # 3. computing out-degree
+        q = """CALL algo.degree({}) YIELD node, degree
+               RETURN node.name, degree
+               ORDER BY node.name"""
+
+        result_set = self.graph.query(q).result_set
+
+        # graph's structure
+        #
+        #      out-degree / in-degree
+        # a0   2            2
+        # a1   2            2
+        # b    5            3
+        # c    2            1
+        # d    0            2
+
+        self.env.assertEqual(result_set[0][0], 'a0')
+        self.env.assertEqual(result_set[0][1], 2)
+
+        self.env.assertEqual(result_set[1][0], 'a1')
+        self.env.assertEqual(result_set[1][1], 2)
+
+        self.env.assertEqual(result_set[2][0], 'b')
+        self.env.assertEqual(result_set[2][1], 5)
+
+        self.env.assertEqual(result_set[3][0], 'c')
+        self.env.assertEqual(result_set[3][1], 1) # multi-edge
+
+        #self.env.assertEqual(result_set[4][0], 'd')
+        #self.env.assertEqual(result_set[4][1], 0)
+
+    def test_all_nodes_all_edges_outdegree(self):
+        q = """CALL algo.degree({dir:'outgoing'}) YIELD node, degree
+               RETURN node.name, degree
+               ORDER BY node.name"""
+
+        result_set = self.graph.query(q).result_set
+
+        # graph's structure
+        #
+        #      out-degree / in-degree
+        # a0   2            2
+        # a1   2            2
+        # b    5            3
+        # c    2            1
+        # d    0            2
+
+        self.env.assertEqual(result_set[0][0], 'a0')
+        self.env.assertEqual(result_set[0][1], 2)
+
+        self.env.assertEqual(result_set[1][0], 'a1')
+        self.env.assertEqual(result_set[1][1], 2)
+
+        self.env.assertEqual(result_set[2][0], 'b')
+        self.env.assertEqual(result_set[2][1], 5)
+
+        self.env.assertEqual(result_set[3][0], 'c')
+        self.env.assertEqual(result_set[3][1], 1) # multi-edge
+
+        # 0 is implicit
+        #self.env.assertEqual(result_set[4][0], 'd') 
+        #self.env.assertEqual(result_set[4][1], 0)
```

Comment:
delete?

Reply:
Nope, this is done on purpose to make sure it is clear that degree 0 is implicit.